### PR TITLE
Spatial Hash Grid Condition

### PR DESCRIPTION
**Pull request by sinfulspartan**
Updated the hash grid condition for huge performance gains that were expected before.

### CODE REVIEW

**Comment by FirstGearGames** [comment]
Thanks. I'm hoping to get this ready within a couple weeks.

File: Assets/FishNet/Runtime/Observing/Conditions/HashGridComponent.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        }
+        IEnumerator UpdateGrid()
+        {
+            //Wait for the HashGrid to be set.
+            while (StaticHashGrid == null)
+                yield return null;
+
+            while (true)
+            {
+                yield return new WaitForSeconds(_updateFrequency);
```
Comment (kamyker) [inline Assets/FishNet/Runtime/Observing/Conditions/HashGridComponent.cs:50]: This should be cached to not allocate

Comment (sinfulspartan) [reply Assets/FishNet/Runtime/Observing/Conditions/HashGridComponent.cs]: Pushed caching of WaitForSeconds.

**Comment by FirstGearGames** [comment]
I appreciate your work but I'm going to close this out for the reasons we discussed about this not fitting into the planned LOD system. If you still want to use a grid system and there's something I can put into FN to make the task easier on yourself let me know.

**Pull request closed without merging**